solver/src/solver/http_solver.rs

```diff
@@ -113,26 +105,23 @@ impl HttpSolver {
             })
             .collect::<HashSet<_>>()
```

Comment:
I wonder if some of this additional collecting could also be removed. Not entirely sure what the chaining business above is all about.

Reply:
`chain`-ing two iterators just causes the second to iterate after the first one is done. Its analogous to `concat` for arrays.

Reply:
Sure but what is it that we are even trying to concatenate here. Shouldn't these be token addresses? Why would anyone ever concatenate them?

Reply:
From the code, it appears that its enumerating over all sell and buy tokens for limit orders as well as pool tokens for AMM pools, and then collecting them into a `HashSet` to generate a list of unique token addresses that can be given to the solver.